internal/armada/server.go

```diff
@@ -28,18 +28,30 @@ import (
 )
 
 func Serve(config *configuration.ArmadaConfig, healthChecks *health.MultiChecker) (func(), *sync.WaitGroup) {
+
+	// TODO Using an error group would be better.
+	// Since we want to shut down everything in a controlled manner in case of an irrecoverable error.
+	// Doesn't look like wg.Done() is called anywhere in this method.
```

Comment:
`wg.Done()` is called in `grpcCommon.Listen` below

Reply:
Removed that part of the comment.